index.bs

```diff
@@ -346,7 +350,190 @@ For example,
+    tricks (or their equivalents in C++/Rust/etc.) whenever reasonable.</p>
+</div>
+
+<h3 id="attributes-like-data">Attributes should behave like data properties</h3>
+
+[[!WEBIDL]] attributes are reified in JavaScript as accessor properties, i.e. properties with
+separate getter and setter functions which can react independently. This is in contrast to the
+"default" style of JavaScript properties, data properties, which do not have configurable behavior
+but instead can simply be set and retrieved, or optionally marked read-only so that they cannot be
+set.
+
+Data property semantics are what are generally expected by JavaScript developers when interfacing
+with objects. As such, although getters and setters allow infinite customizability when defining
+your Web IDL attributes, you should endeavor to make the resulting accessor properties behave as
+much like a data property as possible. Specific guidance in this regard includes:
+
+* Getters must not have any (observable) side effects.
+* Getters should not perform any expensive operations. (A notable failure of the platform in this
+    regard is getters like <code highlight="js">offsetTop</code> performing layout; do not repeat this mistake.)
+* Ensure that your attribute's getter returns the same object each time it is called, until some
+    occurrence in another part of the system causes a logical "reset" of the property's value. In
+    particular, <code highlight="js">obj.attribute === obj.attribute</code> must always hold, and so returning a
+    new value from the getter each time is not allowed.
+* Whenever possible, preserve values given to the setter for return from the getter. That is,
+    given <code highlight="js">obj.attribute = x</code>, a subsequent <code highlight="js">obj.attribute === x</code> should be
+    true. (This will not always be the case, e.g. if a normalization or type conversion step is
+    necessary, but should be held as a goal for normal code paths.)
+
+<h3 id="live-vs-static">Consider whether objects should be live or static</h3>
+
+Objects returned from functions, attribute getters, etc.,
+can either be live or static.
+A <dfn>live object</dfn> is one that continues to reflect
+changes made after it was returned to the caller.
+A <dfn>static object</dfn> is one that reflects
+the state at the time it was returned.
+
+Objects that are the way state is mutated are generally live.
+For example, DOM nodes are returned as live objects,
+since they are the API through which attributes are set and other changes
+to the DOM are made.
+They also reflect changes to the DOM made in other ways
+(such as through user interaction with forms).
+
+Objects that represent a collection
+that might change over time
+(and that are not the way state is mutated)
+should generally be returned as static objects.
+This is because it is confusing to users of the API when
+a collection changes while being iterated.
```

Comment:
This is one of the reasons. The other is that live collections incur a cost at a distance. They make other methods that affect that collection slower (e.g. adding/removing items), which is pretty unexpected.